abel/tools/polar.py

```diff
@@ -14,127 +14,136 @@
-    "origin" is a tuple of (x0, y0) relative to the bottom-left image corner,
-    and defaults to the center of the image.
+    Reprojects a 2D numpy array (**data**) into a polar coordinate system,
+    with the pole placed at **origin** and the angle measured clockwise from
+    the upward direction. The resulting array has rows corresponding to the
+    radial grid, and columns corresponding to the angular grid.
 
     Parameters
     ----------
     data : 2D np.array
-    origin : tuple
-        The coordinate of the image center, relative to bottom-left
-    Jacobian : boolean
-        Include ``r`` intensity scaling in the coordinate transform. 
-        This should be included to account for the changing pixel size that 
+        the image array
+    origin : tuple or None
+        (row, column) coordinates of the image origin. If ``None``, the
+        geometric center of the image is used.
+    Jacobian : bool
+        Include `r` intensity scaling in the coordinate transform.
+        This should be included to account for the changing pixel size that
         occurs during the transform.
     dr : float
-        Radial coordinate spacing for the grid interpolation
-        tests show that there is not much point in going below 0.5
-    dt : float
-        Angular coordinate spacing (in radians)
-        if ``dt=None``, dt will be set such that the number of theta values
-        is equal to the maximum value between the height or the width of 
-        the image.
+        radial coordinate spacing for the grid interpolation.
+        Tests show that there is not much point in going below 0.5.
+    dt : float or None
+        angular coordinate spacing (in radians).
+        If ``None``, the number of angular grid points will be set to the
+        largest dimension (the height or the width) of the image.
 
     Returns
     -------
     output : 2D np.array
-        The polar image (r, theta)
+        the polar image (r, theta)
     r_grid : 2D np.array
         meshgrid of radial coordinates
     theta_grid : 2D np.array
-        meshgrid of theta coordinates
-        
+        meshgrid of angular coordinates
+
     Notes
     -----
-    Adapted from: 
-    http://stackoverflow.com/questions/3798333/image-information-along-a-polar-coordinate-system
-    
-    """
-    # bottom-left coordinate system requires numpy image to be np.flipud
-    data = np.flipud(data)
+    Adapted from:
+    https://stackoverflow.com/questions/3798333/image-information-along-a-polar-coordinate-system
 
+    """
     ny, nx = data.shape[:2]
     if origin is None:
-        origin = (nx//2, ny//2) 
+        origin = (ny // 2, nx // 2)
+    else:
+        origin = list(origin)
+        # wrap negative coordinates
+        if origin[0] < 0:
+            origin[0] += ny
+        if origin[1] < 0:
+            origin[1] += nx
 
     # Determine that the min and max r and theta coords will be...
```

Comment:
There is a typo on this line (I know that you didn't add this, but I just noticed it now). It should say "Determine *what* the min and max..."

Reply:
Corrected.